e2e/framework/provisioner.go

```diff
@@ -0,0 +1,92 @@
+package framework
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"os"
+
+	capi "github.com/hashicorp/consul/api"
+	napi "github.com/hashicorp/nomad/api"
+	vapi "github.com/hashicorp/vault/api"
+)
+
+// ProvisionerOptions defines options to be given to the Provisioner when calling
+// ProvisionCluster
+type ProvisionerOptions struct {
+	Name         string
+	ExpectConsul bool // If true, fails if a Consul client can't be configured
+	ExpectVault  bool // If true, fails if a Vault client can't be configured
+}
+
+type ClusterInfo struct {
+	ID           string
+	Name         string
+	NomadClient  *napi.Client
+	ConsulClient *capi.Client
+	VaultClient  *vapi.Client
+}
+
+// Provisioner interface is used by the test framework to provision a Nomad
+// cluster for each TestCase
+type Provisioner interface {
+	ProvisionCluster(opts ProvisionerOptions) (*ClusterInfo, error)
+	DestroyCluster(clusterID string) error
+}
+
+// DefaultProvisioner is a noop provisioner that builds clients from environment
+// variables according to the respective client configuration
+var DefaultProvisioner Provisioner = new(singleClusterProvisioner)
+
+type singleClusterProvisioner struct{}
+
+func (p *singleClusterProvisioner) ProvisionCluster(opts ProvisionerOptions) (*ClusterInfo, error) {
+	// Build ID based off given name
+	h := md5.New()
+	h.Write([]byte(opts.Name))
+	info := &ClusterInfo{
+		ID:   hex.EncodeToString(h.Sum(nil))[:8],
+		Name: opts.Name,
+	}
+
+	// Nomad client is required
+	if len(os.Getenv("NOMAD_ADDR")) == 0 {
```

Comment:
Can this assume the default of 127.0.0.1:4646 if not set? for even lesser setup to run the tests.

Reply:
Great idea!

This begs the question, should we do the same for vault and consul clients?

Reply:
The default clients for Consul/Vault/Nomad already use the default port and localhost, and read from the env vars if present. But your early return in this block was preventing that. Removing this validation block should make it consistent for all three